standards/source-control.md

```diff
@@ -36,12 +38,42 @@ Once a branch is no longer needed it should be deleted, this helps keep the repo
+
+|Repository Type | Archive Duration | Recovery after Deletion
+|:-:| - | - |
+| Public | 12 months | 90 days |
+| Private | 6 months | 90 days |
+
+Archiving of repositories should be completed following agreement with the relevant parties.
+
 ## Licensing
 
 Repositories should be [clearly named]({{ '/standards/naming-things' | relative_url }}),
 and have an [appropriate licence]({{ '/standards/licencing-software-or-code' | relative_url }})
 and enough documentation that someone new can get started with the project.
 
+## Repository Checklist
+
+|Requirement|Description|Additional Notes
+|:-:| - | - |
+|Should|be [clearly named]({{ '/standards/naming-things' | relative_url }})|you should be able to infer a basic understanding of what something does from its name.|
+|Should|be well documented in the [README file]({{ '/standards/source-control#creating-an-effective-readme' | relative_url}})|there should be enough documentation that someone new can get started with the project.|
+|Should|be initialised using the [das-github-template]{:target="_blank"} |(see: [creating a repository from a template]{:target="_blank"}) ensures new repositories include relevant folder structures, files and licence information.|
+|Should|be built and deployed through a single build and release pipeline|improves maintainability.|
+|Should|use [branch protection rules]{:target="_blank"} that require pull requests|notifies team members of code changes and improves code quality and standardisation through code reviews.|
+|Should|use [branch protection rules]{:target="_blank"} that require status checks on commits|helps to maintain code quality by ensuring it is built and tested in the existing pipeline.|
+|Should|use [branch protection rules]{:target="_blank"} that require [SonarCloud Code Analysis]({{ '/development/static-code-analysis/#static-code-analysis' | relative_url }})|helps to maintain code quality by using our chosen cloud native static code analysis tool. |
+|Should|receive [dependabot alerts and security updates]{:target="_blank"}|improves visibility of vulnerabilities in custom code and dependant packages.|
+|Should|assign the das-contributor team write access|ensures the relevant people have access to the repository.|
+|Should|include a webhook to receive notifications of repository vulnerability alerts|increases visibility of security and vulnerability alerts.|
+|Should|use a [gitignore]{:target="_blank"} file |helps to keep the repository free from temporary local files and artifacts.|
+|Should|have an [appropriate licence]({{ '/standards/licencing-software-or-code' | relative_url }})|a software license tells others what they can and can't do with the source code.|
+|Should NOT|be a fork of another repo||
+|Should NOT|be used to store secrets||
+
 ## Creating an Effective Readme
 
 An effective and comprehensive Readme within a repository can help to simplify and accelerate the job new contributors have getting up to speed with a new code-base.
```

Comment:
I'd like to include an example of what good looks like here. 

For example:
- Description of what the repository contains
- Details on how to get started
- Details of any dependencies
- Explanation of configuration required
- Usage examples (more for tools/helper repos)